options/db_options.cc

```diff
@@ -1073,18 +1073,22 @@ bool MutableDBOptions::IsNowOffPeak(SystemClock* clock) const {
   int64_t now;
   if (clock->GetCurrentTime(&now).ok()) {
     constexpr int kSecondsPerDay = 86400;
-    int since_midnight_seconds = static_cast<int>(now % kSecondsPerDay);
+    constexpr int kSecondsPerMinute = 60;
+    int seconds_since_midnight_to_nearest_minute =
+        (static_cast<int>(now % kSecondsPerDay) / kSecondsPerMinute) *
+        kSecondsPerMinute;
     int start_time = 0, end_time = 0;
     assert(
         TryParseTimeRangeString(daily_offpeak_time_utc, start_time, end_time));
+    assert(start_time != end_time);
 
     // if the offpeak duration spans overnight (i.e. 23:30 - 4:30 next day)
     if (start_time > end_time) {
-      return start_time <= since_midnight_seconds ||
-             since_midnight_seconds <= end_time;
+      return start_time <= seconds_since_midnight_to_nearest_minute ||
+             seconds_since_midnight_to_nearest_minute <= end_time;
     } else {
-      return start_time <= since_midnight_seconds &&
-             since_midnight_seconds <= end_time;
+      return start_time <= seconds_since_midnight_to_nearest_minute &&
+             seconds_since_midnight_to_nearest_minute <= end_time;
```

Comment:
This is OK by me, except I think we should explain it as the time range being inclusive, i.e. including both minutes listed as bounds. As a consequence, "00:00-23:59" or "12:34-12:33" means all day. Also, I'm sure someone will notice the extra minute and ask about it.

Reply:
I think we are spending way too much time on the discussion.  I think saying that the values are inclusive works.  Documenting that 0:00-23:59 means an entire day and 6:00-6:00 means one minute would be satisfactory to me.